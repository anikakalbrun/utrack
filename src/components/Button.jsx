import React from 'react';
import {string, bool, func} from 'prop-types';

Button.propTypes = {
    text: string,
    type: string,
    btnClass: string,
    handleClick: func,
    disabled: bool
};

Button.defaultProps = {
    disabled: false,
    btnClass: '',
    type: "button"
};

export default function Button({text, type, btnClass, handleClick, disabled}) {
    return (
        <button type={type}
                className={`btn ${btnClass}`}
                onClick={handleClick}
                disabled={disabled}
        >{text}</button>
    );
}