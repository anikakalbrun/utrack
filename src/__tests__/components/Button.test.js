import React from 'react';
import Button from '../../components/Button';

let wrapper;
const handleClick = jest.fn();
const props = {
    text: "Submit",
    type: 'button',
    btnClass: "success",
    handleClick,
    disabled: undefined
};

describe('Button', () => {

    it('renders correctly', () => {
        wrapper = renderer.create(<Button {...props}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('button click calls handleClick prop', () => {
        wrapper = shallow(<Button {...props}/>);
        wrapper.find('button').at(0).simulate('click');
        expect(handleClick).toHaveBeenCalled();
    });
});
