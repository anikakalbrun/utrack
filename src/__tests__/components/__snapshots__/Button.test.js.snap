// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className="btn success"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Submit
</button>
`;
